File: frontend/src/types/billing.ts
export type PlanTier = "free" | "basic" | "pro" | "enterprise";
export type SubscriptionStatus = "active" | "cancelled" | "past_due" | "trialing" | "inactive";

export interface Subscription {
  id: string;
  plan: PlanTier;
  status: SubscriptionStatus;
  monthly_voice_limit: number;
  monthly_generation_limit: number;
  voices_used: number;
  generations_used: number;
  current_period_start: string | null;
  current_period_end: string | null;
}

export interface CreateCheckoutSessionRequest {
  plan: PlanTier;
  is_annual?: boolean;
  success_url: string;
  cancel_url: string;
}

export interface CheckoutSessionResponse {
  session_id: string;
  url: string;
}

File: backend/schemas/billing.py
from pydantic import BaseModel
from uuid import UUID
from datetime import datetime


class CreateCheckoutSessionRequest(BaseModel):
    plan: str
    is_annual: bool = False
    success_url: str
    cancel_url: str


class CheckoutSessionResponse(BaseModel):
    session_id: str
    url: str


class SubscriptionResponse(BaseModel):
    id: UUID
    plan: str
    status: str
    monthly_voice_limit: int
    monthly_generation_limit: int
    voices_used: int
    generations_used: int
    current_period_start: datetime | None
    current_period_end: datetime | None

    class Config:
        from_attributes = True


class WebhookPayload(BaseModel):
    type: str
    data: dict
